ray_on_golem/client/client.py

```diff
@@ -192,3 +203,62 @@ def _make_request(
             raise RayOnGolemClientValidationError(
                 "Couldn't validate response data",
             ) from e
+
+    def _start_webserver(self) -> None:
+        with cli_logger.group("Ray On Golem webserver"):
+            if self._is_webserver_running():
+                cli_logger.print("Webserver is already running")
+                return
+
+            cli_logger.print("Starting webserver...")
+            subprocess.Popen(
+                [RAY_ON_GOLEM_PATH, "-p", str(self._base_url.port), "--self-shutdown"],
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                start_new_session=True,
+            )
+
+            for _ in range(10):
+                sleep(2)
+
+                if self._is_webserver_running():
+                    cli_logger.print("Starting webserver done")
+                    return
+
+                cli_logger.print("Webserver is not yet running, waiting additional 2 seconds...")
+
+            cli_logger.abort("Starting webserver failed!")
+
+    def _stop_webserver(self) -> None:
+        with cli_logger.group("Ray On Golem webserver"):
+            if not self._is_webserver_running():
+                cli_logger.print("Webserver is already stopped")
+                return
+
+            cli_logger.print("Requesting webserver stop...")
+
+            response = self._make_request(
+                url=settings.URL_SELF_SHUTDOWN,
+                request_data=models.SelfShutdownRequestData(),
+                response_model=models.SelfShutdownResponseData,
+                error_message="Couldn't send self shutdown request",
+            )
+
+            if response.shutdown_state == ShutdownState.NOT_ENABLED:
+                cli_logger.print("No need to stop webserver, as it was ran externally")
+                return
+            elif response.shutdown_state == ShutdownState.CLUSTER_NOT_EMPTY:
+                cli_logger.print("No need to stop webserver, as cluster is not empty")
+                return
+
+            cli_logger.print("Requesting webserver done, will stop soon")
+
+    def _is_webserver_running(self) -> bool:
+        try:
+            response = self._session.get(
+                str(self._base_url / URL_HEALTH_CHECK.lstrip("/")), timeout=2
```

Comment:
I'd define this timeout also as some constant on a module level... 

(or, we could have them as parameters provided to the class's constructor)